### ISSUE TITLE
Priority Hints API with AMP-HTML

### ISSUE THREAD

**Issue by ghost** [WICG/priority-hints#42]
Not sure if this is useful or not, but I requested the AMP team to whitelist your attribute while I beta test it. See the github issue: https://github.com/ampproject/amphtml/issues/21945

I know your spec says we can add the importance attribute to:

- scipts
- styles
- images
- link rel-preload
- and later iFrames

Would you ever consider adding it to `<amp-img>` as well so we can use this API on the images in the AMP-HTML as well.

**Comment by domfarolino** [comment]
I assume you're interested in having `importance` appear in the list: https://amp.dev/documentation/components/amp-img#attributes correct? If this is the case, that's actually going to be entirely up to AMP, as `amg-img` is a custom element. AMP would be responsible for honoring the `importance` attribute and plumbing its value down to the platform-level where the image request is actually made under-the-hood. Does this make sense?

Basically `importance` is spec'ed for native platform primitives (like `<link>`, `<script>`, etc), and any library or framework that wants to use it can do so in the way they see fit.

**Comment by ayumihamsaki** [comment]
You are correct what you have said. I just thought that if the AMP Team do accept whitelisting it, you may consider also mentioning it in the spec. Right now the spec only talks about native platforms and I thought I open this issue to suggest maybe in the near future adding an example with an amp-html element like `amp-img` - only if the amp team are happy to include it.

Because AMP-HTML is such a large project and on many websites it might give value to users and developers if it did get included in the spec.

I leave it to you to decide, this is just some feedback while I'm beta testing it.

**Comment by ghost** [comment]
You are correct what you have said. I just thought that if the AMP Team do accept whitelisting it, you may consider also mentioning it in the spec. Right now the spec only talks about native platforms and I thought I open this issue to suggest **maybe in the near future adding an example in the spec showing an amp-html element like `amp-img` with the importance attribute** - only if the amp team are happy to include it. e.g.

```html
<script async src="https://cdn.ampproject.org/v0.js" importance="high"></script>

<amp-img src="/static/samples/img/amp.jpg"
  width="1080"
  height="610"
  layout="responsive"
  alt="AMP"
  importance="high"></amp-img>
```

Because AMP-HTML is such a large project and on many websites it might give value to users and developers if it did get included in the spec.

I leave it to you to decide, this is just some feedback while I'm beta testing it.

**Comment by pmeenan** [comment]
The HTML spec doesn't include details about custom element interfaces and frameworks. Those docs best belong with the framework themselves.
